runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -4423,7 +4408,7 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
          clientSession->getSequencingMonitor()->exit();
          }
       abortCompilation = true;
-      if (!enableJITaaSPerCompConn)
+      if (!enableJITaaSPerCompConn || e.getType() == JITServer::MessageType::connectionTerminate)
```

Comment:
Below, for `clientTerminate` case I would like to know if we should close the connection as well. This message is sent by an application thread and that thread is not going to send any other message.

Reply:
Good catch, we should close the connection for `clientTerminate` as well.